### PR TITLE
Add Temporal support

### PR DESCRIPTION
**Pull request by sffc**
This started by adding a few toLocaleString()s, and then I noticed that you were doing a bunch of stuff with dates, so I decided to add in the Temporal polyfill.  We're trying to test out Temporal in various types of projects to figure out its ergonomics. :)

I have package.json pointing at my Temporal fork, pending https://github.com/tc39/proposal-temporal/pull/516

I added a parsing function in utils.mjs, maybe to be replaced by https://github.com/tc39/proposal-temporal/issues/515

### CODE REVIEW

**Review by pipobscure — APPROVED** [review]
I like the direct installability for modern node versions. It can only help.

(Sorry wrong place for comment)

File: lib/util.mjs [hunk 1]
```diff
@@ -15,6 +15,7 @@ limitations under the License.
 */
 
 import { average } from './math.mjs';
+import { Temporal } from "proposal-temporal/polyfill/lib/index.mjs";
```
Comment (MylesBorins) [inline lib/util.mjs]:
```suggestion
import { Temporal } from 'proposal-temporal';
```

I'm p sure this should work, will check locally.

File: lib/util.mjs [hunk 2]
```diff
@@ -47,8 +48,31 @@ function printWindows(windowTitle, windows, totalLength = 30, windowLength = DEF
...
+function parseMDYToTemporal(mdyString) {
+  // Step 1: parse the date using the JS Date constructor.
+  // jsDate is set to midnight on the parsed date in my current time zone.
+  const jsDate = new Date(mdyString);
+
+  // Step 2: convert to a Temporal.Absolute.
+  const absolute = Temporal.Absolute.fromEpochMilliseconds(jsDate);
+
+  // Step 3: convert to a Temporal.DateTime using my current time zone.
+  const dateTime = absolute.inTimeZone(Temporal.now.timeZone());
```
Comment (MylesBorins) [inline lib/util.mjs:62]: So curious. The data we are working with and showing is very specific to NYC, and I think show the data as of NYC makes a lot of sense. is there a best practice here when it comes to INTL?

**Comment by MylesBorins** [comment]
Also, in case it wasn't clear. This is super cool and I really appreciate the PR here.

File: lib/util.mjs (hunk 2 above)
Comment (sffc) [reply lib/util.mjs:62]: The problem is that the CSV file has date formats like "3/11/20", which is recognized by the JavaScript Date constructor but not by Temporal.  However, JavaScript Date needs a time component, and when parsing a date without a time, it sets the time to midnight in the system time zone ([spec](https://tc39.es/ecma262/#sec-date-year-month-date-hours-minutes-seconds-ms)):

> Set O.[[DateValue]] to TimeClip(UTC(finalDate)).

where UTC "converts t from local time to UTC".

When converting to Temporal, I need to undo what the Date constructor did, which is why I passed in the system time zone here.

However, in general, better practice might be to write a custom parser that extracts the year, month, and day out of the y/m/d format, and then pass the results into Temporal.Date.from().  Then you don't depend on the fragile behavior of the JavaScript Date constructor.

**Pull request closed without merging**